src/engine/SpatialJoin.cpp

```diff
@@ -433,14 +440,14 @@ Result SpatialJoin::baselineAlgorithm() {
       // Ensure `maxResults_` constraint using priority queue
       intermediate.push(std::pair{rowRight, dist.getInt()});
       // Too many results? Drop the worst one
-      if (intermediate.size() > static_cast<size_t>(maxResults_)) {
+      if (intermediate.size() > static_cast<size_t>(maxResults.value())) {
```

Comment:
this line strongly suggests, that `maxResults`  should be a ` size_t ` in the first place